Resources/public/js/views/ez-languageselectionboxview.js

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-languageselectionboxview', function (Y) {
+    "use strict";
+    /**
+     * Provides the language selection box view class
+     *
+     * @method ez-languageselectionboxview
+     */
+    Y.namespace('eZ');
+
+    var CONFIRM = 'confirm',
+        CANCEL = 'cancel';
+
+    /**
+     * The language selection box view.
+     *
+     * @namespace eZ
+     * @class LanguageSelectionBoxView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.LanguageSelectionBoxView = Y.Base.create('languageSelectionBoxView', Y.eZ.TemplateBasedView, [], {
+        BASE_TRANSLATION_VISIBLE_CLASS: 'is-visible',
```

Comment:
and `is-visible` and `is-selected` are a bit too generic should be something like `is-base-translation-visible` and `is-translation-selected`.